Components/Queues/PhpAmqpLibQueueManager.php

```diff
@@ -0,0 +1,309 @@
+            }
+
+            if (!$this->connection) {
+                throw new \RuntimeException(sprintf('Unable to connect to any of the following hosts:%s%s', PHP_EOL, implode(PHP_EOL, $tested)));
+            }
+        }
+
+        if (!$this->channel) {
+            $this->declareChannel();
+        }
+    }
+
+    /**
+     * Destroys the connection with the queuing system.
+     */
+    public function disconnect()
+    {
+        try {
+            if ($this->isConnected()) {
+                $this->connections[0]->close();
+            }
+            $this->connection = null;
+        } catch (\Exception $exception) {
+            throw new \Exception($exception->getMessage());
+        }
+    }
+
+    /**
+     * Returns true if a connection already exists with the queueing system, false otherwise.
+     * @return bool
+     */
+    public function isConnected(): bool
+    {
+        return $this->connections[0]  ? $this->connections[0]->isConnected() : false;
+    }
+
+    /**
+     * Declares the queue to drive the message
+     *
+     * @param string $name      The name of the que
+     * @param int    $flag      See AMQPQueue::setFlags()
+     * @param array  $arguments See AMQPQueue::setArguments()
+     *
+     * @return \AMQPQueue
+     *
+     * @throws \AMQPChannelException
+     * @throws \AMQPConnectionException
+     * @throws \AMQPQueueException
+     */
+    public function declareQueue(string $name, int $flag = AMQP_DURABLE, array $arguments = [])
+    {
+        if (!$this->channel) {
+            throw new \AMQPChannelException('There is no channel available');
+        }
+
+        $this->queue = $name;
+        $durable = $flag === AMQP_DURABLE ? true : false;
+        return $this->channel->queue_declare($this->queue, false, $durable, false, false, false, new AMQPTable([$arguments]));
+    }
+
+    /**
+     * Publish a message
+     * @param AMQPMessage $message
+     */
+    public function publish(AMQPMessage $message)
+    {
+        $this->channel->basic_publish($message, self::EXCHANGE_NAME, $this->queue);
+    }
+
+    /**
+     * Add a connection to the browker
+     * @param $connection
+     * @return $this
+     */
+    public function addConnection($connection): self
+    {
+        $this->connections[] = $connection;
+
+        return $this;
+    }
+
+    /**
+     * Returns all the connections available
+     * @return \AMQPConnection[]|array|null
+     */
+    public function getConnections():? array
+    {
+        return $this->connections;
+    }
+
+    /**
+     * Catch a channel inside the connection
+     * @return AMQPChannel
+     */
+    public function declareChannel(): AMQPChannel
+    {
+        $this->channel = reset($this->connections)->channel();
+        $this->channel->basic_qos(null, self::PREFETCH_COUNT, null);
+        return $this->channel;
+    }
+
+    /**
+     * AMQP Exchange is the publishing mechanism
+     */
+    public function declareExchange($exchange): void
```

Comment:
This method doesn't seem to be used in the code, there is only 1 reference in ```Components/Queues/Drivers/PhpAmqpLibDriver.php``` method ```setExchange``` but that method is not used anywhere

Reply:
This method is not used inside the EAI scope because we don't use a specified exchange. But, I leave this method in case of need to declare some specific one.